src/zope/file/testing.py

```diff
@@ -146,3 +136,31 @@ def _extractContentInfo(self, item):
         info['object'] = obj
         info['url'] = urllib.quote(oid.encode('utf-8'))
         return info
+
+@implementer(IBrowserPublisher)
+class ManagementViewSelector(BrowserView):
+    """View that selects the first available management view.
+
+    Support 'zmi_views' actions like: 'javascript:alert("hello")',
+    '../view_on_parent.html' or '++rollover++'.
+    """
+    # Copied from zope.app.publication
+
+    def browserDefault(self, request):
+        return self, ()
+
+    def __call__(self):
+        item = getFirstMenuItem('zmi_views', self.context, self.request)
+
+        if item:
+            redirect_url = item['action']
+            if not (redirect_url.startswith('../') or \
+                    redirect_url.lower().startswith('javascript:') or \
+                    redirect_url.lower().startswith('++')):
```

Comment:
That was copied directly as-is from zope.app.publication (because it doesn't support Py3 yet). Eventually we should be able to remove this class entirely. But I can simplify more.

PEP8 also says that 'or' belongs at the beginning of the line now. (I was glad when they made that change.)